ipwhois/net.py

```diff
@@ -579,6 +583,7 @@ def lookup_asn(self, retry_count=3):
         # Attempt to resolve ASN info via Cymru. DNS is faster, try that first.
         try:
 
+            self.resolver.lifetime = self.resolver.timeout * retry_count
```

Comment:
Needs to account for if retry_count == 0. Change to (keep the new line for char limit):
```python
self.dns_resolver.lifetime = self.dns_resolver.timeout * (
    retry_count if retry_count > 0 else 1)
```